components/pdfviewer/events.md

```diff
@@ -0,0 +1,133 @@
+
+The `OnError` event fires when a file error occurs. For example, the user tries to open a corrupt file or a file that is not in the correct format.
+
+The event handler receives an argument of type [`PdfViewerErrorEventArgs`](/blazor-ui/api/Telerik.Blazor.Components.PdfViewerErrorEventArgs), which exposes a `Message` property. See the [example below](#example).
+
+
+## OnOpen
+
+The `OnOpen` event fires when the user selects a file to open from the [PDF Viewer toolbar]({%slug pdfviewer-toolbar%}).
+
+The event handler receives an argument of type [`PdfViewerOpenEventArgs`](/blazor-ui/api/Telerik.Blazor.Components.PdfViewerOpenEventArgs). The event is cancellable and allows the application to obtain the PDF file name, size and contents as a `Stream`. The `OnOpen` handler is not required to open a file. See the [example below](#example).
+
+
+## ZoomChanged
+
+The `ZoomChanged` event fires when the user clicks on the zoom in/out buttons, or selects a new zoom level from the ComboBox.
+
+The event handler receives the new zoom level as an argument of type `double`. To apply the new zoom level, set it as a new `Zoom` parameter value. Not setting it will effectively cancel the event.
+
+
+## Example
+
+>caption Handle or cancel Blazor PDF Viewer Events
+
+````CSHTML
+<p> Last event: @EventLog </p>
+
+<p><label> <TelerikCheckBox @bind-Value="@AllowDownloads" /> Allow Downloads </label></p>
+
+<TelerikPdfViewer Data="@PdfSource"
+                  Height="600px"
+                  OnDownload="@OnPdfDownload"
+                  OnError="@OnPdfError"
+                  OnOpen="@OnPdfOpen"
+                  Zoom="@PdfZoom"
+                  ZoomChanged="@OnPdfZoomChanged">
+</TelerikPdfViewer>
+
+@code {
+    private byte[] PdfSource { get; set; }
+
+    private double PdfZoom { get; set; } = 1.25;
+
+    private bool AllowDownloads { get; set; } = true;
+
+    private string EventLog { get; set; } = "...";
+
+    private async Task OnPdfDownload(PdfViewerDownloadEventArgs args)
+    {
+        if (AllowDownloads)
+        {
+            args.FileName = "PDF-Viewer-Download";
+            EventLog = $"Download {args.FileName}.pdf";
+        }
+        else
+        {
+            args.IsCancelled = true;
+            EventLog = $"Download cancelled";
+        }
+    }
+
+    private async Task OnPdfError(PdfViewerErrorEventArgs args)
+    {
+        // To trigger the event, rename a random file to error.pdf and try to open it.
+        EventLog = "Error: " + args.Message;
+    }
+
+    private async Task OnPdfOpen(PdfViewerOpenEventArgs args)
+    {
+        var file = args.Files.FirstOrDefault();
+
+        if (file.Size > 1024 * 1024)
+        {
+            args.IsCancelled = true;
+            EventLog = $"Open cancelled conditionally. File {file.Name} ({file.Size} bytes) is larger than 1 MB.";
+        }
+        else
+        {
+            // Get the PDF file contents if necessary.
+            var buffer = new byte[file.Stream.Length];
```

Comment:
If we reach this point, the file is not opened. Is this expected?

Reply:
(SignalR message size)